br/pkg/lightning/backend/local/region_job.go

```diff
@@ -788,11 +789,8 @@ func (h *regionJobRetryHeap) Pop() any {
 // a heap of jobs internally based on the regionJob.waitUntil field.
 type regionJobRetryer struct {
 	// lock acquiring order: protectedClosed > protectedQueue > protectedToPutBack
```

Comment:
note that we should have `protectedClosed` locked when accessing `protectedQueue`

Reply:
good point @lance6716